### DISCUSSION TITLE
Drop snap for Flatpak (or an AppImage) (for Linux)

### DISCUSSION THREAD

**Discussion by helium18** [massCodeIO/massCode#60] [Ideas]
As the title suggests. Snap should be dropped for Flatpak support. Flatpak runs on all gnu/linux distributions and is completely open sourced. Snaps also perform horribly. There's a general dislike for snap among linux users.

It becomes a huge barrier for me to install this without build instructions and then seeing that snap is the only installation medium, and I'm sure that there are other people like me on this platform, who'd like to use this app but are limited by the installation mediums. 

I think, with flatpak OR an appimage (or both preferably) massCode will be able to target all linux users.

**Comment by antonreshetov** [comment]
Unfortunately, I am a maс user and can not test all platforms. I tried to make a build to `.AppImage`, `.deb`, but when I start the application just a white screen.  At the same time `snap` works.

**Reply by CRBroughton** [reply]
I've managed to compile both an AppImage and a .deb package on my Kubuntu 20.04 machine. Is there any way I can contribute (or upload) the compiled executables? I simply made these changes:

```typescript
  linux: {
    target: ['snap', 'deb', 'AppImage'],
    icon: 'config/icons',
    maintainer: 'masscodeIO'
  },
```

Wouldn't mind compiling these two specific Linux builds for future releases as well.

Edit: It's also possible you could do this through Github Actions, although I haven't personally looked into it.

**Reply by antonreshetov** [reply]
Actually there is no other way like the one you mentioned to make a build. As I said before, I made builds, but they did not work correctly in VM Ubuntu 20.

**Reply by CRBroughton** [reply]
Here is the debian package running on my Kubuntu machine. As you can see it does compile correctly and runs, and the extension does connect to the application.

![image](https://user-images.githubusercontent.com/71838537/166112031-712c5b6f-ed2b-4658-9521-f33e2487b33d.png)

**Reply by antonreshetov** [reply]
@CRBroughton That's cool. If someone would test it on Ubuntu that would be great. I just don't want to include a broken build in the release.

**Reply by CRBroughton** [reply]
That totally understandable :) I've compiled both binaries and have put them on my fork here:

https://github.com/CRBroughton/massCode/releases/tag/v2.5.0

@helium18 Would you be willing to verify if these both work for you? I can also look into automating the process using Github Actions when I have some time.

**Reply by grzesiek1owline** [reply]
@CRBroughton thanks! deb file works fine on PopOs (distro based on Ubuntu).

**Reply by grzesiek1owline** [reply]
@CRBroughton can You create appImage from newest version? ;>

**Reply by CRBroughton** [reply]
@grzesiek1owline there are now auto-built ones here:

https://github.com/CRBroughton/massCode/releases

**Reply by agmeiner-bhs** [reply]
@CRBroughton: Did you stopp the auto-built? think there is still the need for flatpaks...

**Reply by CRBroughton** [reply]
@AndreasGmeinerBHS I did; I'm no longer using the application. The requirements above to add additional package types like appimage are simple enough. I can't remember if I had trouble with the last attempt to bundle it to an appimage, but that might of been partially the reason I stopped doing so.

**Comment by antonreshetov** [comment]
I started the discussion here - #87

**Comment by mbnoimi** [comment]
+1 
Flatpak or/and Appimage
